R/utilities-qualification-gof.R

```diff
@@ -0,0 +1,409 @@
+#' @title getQualificationGOFPlotData
+#' @description Build dataframes and metadata for each GOF plot
+#' @param configurationPlan A `ConfigurationPlan` object
+#' @return plotGOFdata, a list of lists of the form list(dataframe,metadata) specific to each GOF plot
+getQualificationGOFPlotData <- function(configurationPlan) {
+  plotGOFdata <- list()
+  for (plt in seq_along(configurationPlan$plots$GOFMergedPlots)) {
+    gofPlotConfiguration <- configurationPlan$plots$GOFMergedPlots[[plt]]
+
+
+    plotGOFDataframe <- NULL
+    plotGOFMetadata <- list()
+    plotGOFMetadata$title <- gofPlotConfiguration$Title
+    plotGOFMetadata$sectionID <- gofPlotConfiguration$SectionId
+    plotGOFMetadata$plotTypes <- ospsuite::toPathArray(gofPlotConfiguration$PlotType)
+
+
+    plotGOFMetadata$axesSettings <- lapply(plotGOFMetadata$plotTypes, function(pltType) {
+      getAxesSettings(configurationPlan$plots$AxesSettings[[gofPlotAxesSettings[[pltType]]]])
+    })
+    names(plotGOFMetadata$axesSettings) <- plotGOFMetadata$plotTypes
+
+    plotGOFMetadata$groups <- list()
+
+
+    for (group in seq_along(gofPlotConfiguration$Groups)) {
```

Comment:
If we do not need to use the index, I suggest we use directly the loop on object itself:
```
for(group in gofPlotConfiguration$Groups){
caption <- group$Caption
symbol <- group$Symbol
outputMappings <- group$OutputMappings
...
}
```

Reply:
Done